src/ParallelAlgorithms/Interpolation/Protocols/ComputeTargetPoints.hpp

```diff
@@ -63,24 +55,36 @@ namespace intrp::protocols {
 struct ComputeTargetPoints {
   template <typename ConformingType>
   struct test {
+    static_assert(std::is_same_v<ConformingType, void>,
+                  "ComputeTargetPoints conforming type must be templated on "
+                  "the InterpolationTargetTag, but it is not. See first "
+                  "template parameter of std::is_same_v for the conforming "
+                  "type and its templates.");
+  };
+
+  template <template <typename...> class ConformingType,
+            typename InterpolationTargetTag, typename... Ts>
+  struct test<ConformingType<InterpolationTargetTag, Ts...>> {
```

Comment:
There's actually an easy enough way around this I think: Either require the conforming type exposes the `InterpolationTargetTag` as a type alias, from which you can then retrieve what you need. Not sure if there's in issue with circular dependencies. Or, make the `temporal_id` a requirement of the `ComputeTargetPoints` protocol itself, so it doesn't have to "reach up" into the InterpolationTargetTag.

Reply:
I like avoiding the `template template` expressions, but this adds the additional requirement that `temporal_id` defined in the `InterpolationTargetTag` be exactly the same as the one defined in `ComputeTargetPoints` (otherwise bad things might happen). This is essentially equivalent to making `ComputeTargetPoints` templated on `InterpolationTargetTag` to ensure this...